LibreNMS/Plugins/DeviceTrait.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace LibreNMS\Plugins;
+
+use App\Models\Device;
+
+trait DeviceTrait
+{
+    protected static $device_view = 'device_overview';
```

Comment:
Hi @mpikzink 
In order to allow different views to be selected, if necessary, I would suggest to replace `$device_view` protected static variable with a function `device_view($device)`. Function, that will, by default, return the 'device_overview' string, like it is now. 
If a plugin decides to override this function, the plugin will be able to call different views, for different devices, with it's own logic. 
(in my case, I have devices that my SupportTeam need to access, and some they shouldnt. Logic is based on name of the device, location, etc etc). 

```
public static function device_view($device)
    {
        return 'menu;
    }
```

Same logic would apply for PortTrait with 2 parameters (device and port).